packages/valory/skills/decision_maker_abci/behaviours/base.py

```diff
@@ -72,6 +72,56 @@ def remove_fraction_wei(amount: int, fraction: float) -> int:
     raise ValueError(f"The given fraction {fraction!r} is not in the range [0, 1].")
 
 
+def perturb_y_if_x_equals_y(x: int, y: int) -> float:
+    """Introduce perturbation to the y value if x equals y."""
+    epsilon = (x + y) / 10000
+    y += epsilon if random.choice([True, False]) else -epsilon
+    return y
+
+
+def calculate_kelly_bet_amount(
+    x: int, y: int, p: float, c: float, b: int, f: float
+) -> int:
+    """Calculate the Kelly bet amount."""
+    if b == 0:
+        return 0
+    if x == y:
+        # o/w kelly traders will never be the first to bet in a new market
+        y = perturb_y_if_x_equals_y(x, y)
```

Comment:
If x==y both denominator AND numerator are zero! So the previous implementation avoided division by zero but didn't solve the issue of 0 bet amount.